tzrec/models/rank_model.py

```diff
@@ -216,24 +211,27 @@ def _loss_impl(
         else:
             raise ValueError(f"loss[{loss_type}] is not supported yet.")
 
-        if sample_weight_name:
-            losses[loss_name] = torch.mean(
-                losses[loss_name] * sample_weights
-            ) / torch.mean(sample_weights)
+        losses[loss_name] = torch.mean(losses[loss_name] * loss_weight)
```

Comment:
- loss_weight should be None rather than ones tensor when do not use loss weight.
- reduction in _init_loss_impl should be none when do not use loss weight.